### PR TITLE
Charts 262: Creating monitoring chart

### PR DESCRIPTION
**Pull request by bmitchinson**

### CODE REVIEW

File: docs/index.yaml
Comment (LtChae) [inline docs/index.yaml:1460]: So nice to see other community helm charts in here

File: urban-os/values.yaml
Comment (LtChae) [inline urban-os/values.yaml:103]: Will this always be prefaced with urban-os? Or is that the release name

File: urban-os/values.yaml
Comment (LtChae) [inline urban-os/values.yaml:305]: The tracer alert mentions a dataset with an id specific to the SCOS instance. This should probably be pulled out to the SCOS repo.

File: urban-os/values.yaml
Comment (LtChae) [inline urban-os/values.yaml:324]: Joomla is not part of the standard urbanos deployment and is SCOS instance specific. This should be pulled out to the SCOS repo.

File: urban-os/values.yaml
Comment (LtChae) [inline urban-os/values.yaml:333]: Kube bench runs are not part of the standard urbanos deployment. This can be removed altogether.

File: urban-os/values.yaml
Comment (LtChae) [inline urban-os/values.yaml:351]: Parking prediction metrics are specific to the SCOS instance. It should be pulled up to the SCOS repo.

File: urban-os/values.yaml
Comment (LtChae) [inline urban-os/values.yaml:232]: This api endpoint is specific to SCOS, though its only excluded here. We could try removing the exclusion and see whether this is still relevant.

Comment (LtChae) [reply urban-os/values.yaml:232]: See it twice below as well

**Review by LtChae — COMMENTED** [review]
Some SCOS instance things snuck in

File: urban-os/values.yaml
Comment (bmitchinson) [reply urban-os/values.yaml:103]: That is dependent on the release name

File: urban-os/values.yaml
Comment (bmitchinson) [reply urban-os/values.yaml]: moved to scos

File: urban-os/values.yaml
Comment (bmitchinson) [reply urban-os/values.yaml]: removed

File: urban-os/values.yaml
Comment (bmitchinson) [reply urban-os/values.yaml]: Moved

File: urban-os/values.yaml
Comment (bmitchinson) [reply urban-os/values.yaml]: Moved

File: urban-os/values.yaml
Comment (bmitchinson) [reply urban-os/values.yaml:232]: Removed `{instance!~".*(/api/v1/predictions)"}` and noticed that it functions the same as prod does currently, so I feel alright about leaving it removed in the public chart, and not accounting for it in `urban-os-deploy`.

removed + pushed up

**Comment by bmitchinson** [comment]
WIP: Do not merge until "monitoring-sub-chart" has been approved + merged into the template repository

File: urban-os/values.yaml
Comment (narruda789) [reply urban-os/values.yaml]: @LtChae Is there a way to reference the release name dynamically?

File: docs/index.yaml
Comment (ksmith-accenture) [inline docs/index.yaml:1476]: Is prometheus and grafana supposed to be in here still?

File: monitoring/Chart.yaml
Comment (ksmith-accenture) [inline monitoring/Chart.yaml:3]: Is this supposed to be "Prometheus and Grafana charts"?

File: monitoring/README.md
Comment (ksmith-accenture) [inline monitoring/README.md:3]: suggested verbiage: "as a unified UrbanOS monitoring solution"

File: monitoring/templates/prometheus-ingress.yaml
Comment (ksmith-accenture) [inline monitoring/templates/prometheus-ingress.yaml:15]: For the sake of cleanliness perhaps this could be called "monitoring-prometheus".  The urban-os nature is implied by it existing in the namespace

File: monitoring/templates/grafana-ingress.yaml
Comment (ksmith-accenture) [inline monitoring/templates/grafana-ingress.yaml:15]: ditto of below

File: monitoring/values.yaml
Comment (ksmith-accenture) [inline monitoring/values.yaml:8]: Having the `alb.ingres...` annotation here hard codes this new monitoring chart to the AWS ingress schema.  I don't know what the solution for this is, so i'm not going to suggest blocking the PR on solving this; just food for thought as this is going to be an issue when we try to deploy on Azure

File: monitoring/values.yaml
Comment (ksmith-accenture) [inline monitoring/values.yaml:12]: Ditto to above

File: monitoring/values.yaml
Comment (ksmith-accenture) [inline monitoring/values.yaml:28]: Ditto to this inadvertently pinning us to AWS, once again not sure if there is a solution

File: urban-os/values.yaml
Comment (bmitchinson) [reply urban-os/values.yaml]: We've changed it so that it will always be prefaced instead of being dynamic, so I think this can be resolved 👍🏼

`fullnameOverride` in the prometheus server values let us call it `urban-os-prometheus-server`, no matter the release

Comment (bmitchinson) [reply urban-os/values.yaml]: This was also accomplished for grafana

File: monitoring/values.yaml
Comment (bmitchinson) [reply monitoring/values.yaml:8]: This has been changed to

```
# A place to define a healthcheck path or other grafana specific annotations
grafanaIngress:
  annotations: {}

# A place to define a healthcheck path or other prometheus specific
# annotations
prometheusIngress:
  annotations: {}
```

With healthchecks defined in the deploy repo. There should no AWS stuff in the public chart, so I think this fixes that but makes it clear that it may need to be filled in.

File: monitoring/values.yaml
Comment (bmitchinson) [reply monitoring/values.yaml:28]: Moved CloudWatch datasource to urban-os-deploy in an effort not to pin us to aws in the public chart

File: monitoring/templates/grafana-ingress.yaml
Comment (bmitchinson) [reply monitoring/templates/grafana-ingress.yaml:15]: changed.

File: monitoring/templates/prometheus-ingress.yaml
Comment (bmitchinson) [reply monitoring/templates/prometheus-ingress.yaml:15]: good call, this has been changed.

File: monitoring/README.md
Comment (bmitchinson) [reply monitoring/README.md:3]: Added

File: monitoring/Chart.yaml
Comment (bmitchinson) [reply monitoring/Chart.yaml:3]: Sure is. Corrected.

File: docs/index.yaml
Comment (bmitchinson) [reply docs/index.yaml:1476]: great catch, not sure why these weren't removed automatically.

regenerating the docs index.yaml no longer puts them there, so i've manually removed.

File: urban-os/templates/prometheus-ingress.yaml
Comment (bmitchinson) [reply urban-os/templates/prometheus-ingress.yaml]: Good idea, added.

File: urban-os/templates/grafana-ingress.yaml
Comment (bmitchinson) [reply urban-os/templates/grafana-ingress.yaml]: Good idea, added.

**Review by LtChae — APPROVED** [review]

**Pull request merged**
